engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/Atom.java

```diff
@@ -203,22 +202,6 @@ boolean hasFunctionalDispatch() {
     return true;
   }
 
-  @ExportMessage
-  boolean isDate(@CachedLibrary("this") InteropLibrary iop) {
```

Comment:
Heuréka, this trick is no longer needed! Instead of Atom representing Date we allocate `EnsoDate` instance which properly supports `isDate` and `asDate` conversion.